e2e/cypress/integration/channel/message_spec.js

```diff
@@ -140,4 +141,63 @@ describe('Message', () => {
             cy.get(divPostId).should('have.class', 'post--system');
         });
     });
+
+    it('MM-16730 Reply to an older message', () => {
+        cy.getCurrentChannelId().then((channelId) => {
+            // # Get yesterdays date in UTC
+            const yesterdaysDate = new Date().getTime() - 86400000;
+
+            // # Post a day old message
+            cy.postMessageAs({sender: sysadmin, message: 'Hello from yesterday', channelId, createAt: yesterdaysDate}).
+                its('id').
+                should('exist').
+                as('yesterdaysPost');
+        });
+
+        // # Add two subsequent posts
+        cy.postMessage('One');
+        cy.postMessage('Two');
+
+        cy.get('@yesterdaysPost').then((postId) => {
+            // # Open RHS comment menu
+            cy.clickPostCommentIcon(postId);
+
+            // # Upload an attachment to the reply
+            cy.fileUpload('#fileUploadInput').wait(TIMEOUTS.TINY);
+
+            // # Reply with the attachment
+            cy.postMessageReplyInRHS('A reply to an older post with attachment');
+
+            // # Get the latest reply post
+            cy.getLastPostId().then((replyId) => {
+                // * Verify that the reply is in the channel view with matching text
+                cy.get(`#post_${replyId}`).within(() => {
+                    cy.get('.post__link').should('be.visible').and('have.text', 'Commented on sysadmin\'s message: Hello from yesterday');
```

Comment:
Haven't looked at the react code yet, but could you update the `.post__link` element to use an id?

Reply:
I will add an ID in for that element. Is there a preferred format? "#postLink" or "#postLink_{childPost.id}".

Reply:
`#postLink` (like you have done) should be fine here. Especially since you are doing it inside a `within`. If we were to want to get the post link without the `within`, I'd say it should be a `data-testid` attribute with the id of the post slapped on it as well. No changes required IMO at the moment though.

Reply:
Cool - I was in there for the latter feedback so adjusted this.